lib/id3c/cli/redcap.py

```diff
@@ -417,6 +417,28 @@ def is_complete(instrument: str, data: dict) -> bool:
     }
 
 
+def completion_status_field(instrument: str) -> str:
+    """
+    Returns the REDCap automatic field name for the completion status of
+    *instrument*.
+
+    If want to know the completion status itself, use :func:`is_complete`
+    instead.
+    """
+    # XXX TODO: It would be good to normalize *instrument* here, including:
+    #
+    #   - Lowercasing
+    #   - Replacing runs of whitespace and/or non-alphanumerics (?) with a
+    #     single underscore.
+    #   - Maybe: removing leading numbers?
+    #
+    # The full set of transformations REDCap applies aren't entirely clear to
+    # me at the moment, so I'm punting for now.  The caller must provide the
+    # internal names.
+    #   -trs, 8 Jan 2020
+    return f"{instrument}_complete"
+
+
```

Comment:
I am a little unclear on the added value of this `completion_status_field` and what you hope to one day achieve in the TODO statement. Would you mind elaborating, please?

Reply:
Sure.

The reason for the function is to centralize and abstract away the transformation, instead of leaving it increasingly sprinkled throughout the code, so that it's easier to update later. Yes, the transformation is trivial right now, but, for example, the `--instrument` option for `id3c redcap-det generate` takes user-provided values and those may not be valid without normalization, which this function could do.

Is this abstraction necessary right now? Nope. But it was ~no extra work, and IMO accomplishes two important things:

1. Increases ease of future changes by centralizing details
2. Improves readability, because now the code talks about _meaningful concepts_ like "completion status" instead of being a list of operations (like "append `_complete` to the name of this instrument")

Reply:
Ah, I see. When you mentioned normalizing `instrument`, for some reason I was thinking that was the field name in REDCap rather than the user provided value. Also, your listed reasons make sense to me. Thanks!